src/ARM.h

```diff
@@ -171,20 +219,48 @@ class ARM
     u32 DataRegion;
     s32 DataCycles;
 
-    u32 R[16]; // heh
+    alignas(64) u32 R[16]; // heh
```

Comment:
Did you mean `u64` here?

Reply:
the alignas? no. i explicitly meant to align it to a host cacheline. which should be 64 bytes. it seemed to give a noticeable performance boost doing so in a few places (though maybe that was just luck?)

Reply:
In that case, I think you might want [`std::hardware_destructive_interference_size` or `std::hardware_constructive_interference_size`](https://en.cppreference.com/w/cpp/thread/hardware_destructive_interference_size).

Reply:
what's the difference between the two?

Reply:
From the linked reference page:

> 1) Minimum offset between two objects to avoid false sharing. Guaranteed to be at least alignof([std::max_align_t](http://en.cppreference.com/w/cpp/types/max_align_t))
> 2) Maximum size of contiguous memory to promote true sharing. Guaranteed to be at least alignof([std::max_align_t](http://en.cppreference.com/w/cpp/types/max_align_t))

It has details and examples.